pkg/devfile/adapters/common/command.go

```diff
@@ -134,17 +147,32 @@ func GetRunCommand(data data.DevfileData, devfileRunCmd string) (runCommand comm
 // ValidateAndGetPushDevfileCommands validates the build and the run command,
 // if provided through odo push or else checks the devfile for devBuild and devRun.
 // It returns the build and run commands if its validated successfully, error otherwise.
-func ValidateAndGetPushDevfileCommands(data data.DevfileData, devfileBuildCmd, devfileRunCmd string) (pushDevfileCommands []common.DevfileCommand, err error) {
+func ValidateAndGetPushDevfileCommands(data data.DevfileData, devfileInitCmd, devfileBuildCmd, devfileRunCmd string) (pushDevfileCommands []common.DevfileCommand, err error) {
 	var emptyCommand common.DevfileCommand
-	isBuildCommandValid, isRunCommandValid := false, false
+	isInitCommandValid, isBuildCommandValid, isRunCommandValid := false, false, false
+
+	initCommand, initCmdErr := GetInitCommand(data, devfileInitCmd)
```

Comment:
can we please add check for `initCmdErr != nil`? and can't we just call it `err`?

Reply:
this will simplify `isInitCmdEmpty && initCmdErr == nil` statement below

Reply:
Currently it checks if `initCmdErr != nil` on line 191, because the errors are only amended if any of the commands are invalid, to match the existing flow.